server/src/test/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettingsTests.java

```diff
@@ -203,4 +203,20 @@ public void testInvalidHighDiskThreshold() {
         assertThat(cause, hasToString(containsString("low disk watermark [85%] more than high disk watermark [75%]")));
     }
 
+    public void testSequenceOfUpdates() {
+        Settings settings = Settings.EMPTY;
+
+        final ClusterSettings clusterSettings = new ClusterSettings(settings, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        new DiskThresholdSettings(settings, clusterSettings); // this has the effect of registering the settings updater
+
+        settings = clusterSettings.applySettings(Settings.builder()
```

Comment:
I see what's happened here, and I must apologise. The unit test I suggested was looking at `applySettings` but really I should have used `updateSettings` since this is where the update is calculated. My bad.

However, this test still passes on `master` without any changes so I don't think it should be added here. Could you make an appropriate test based around `updateSettings` instead, and check that it fails on `master` and passes on this branch?

Reply:
No worries at all. Sure, I'll update the test and run it on this branch and on `master`.